### DISCUSSION TITLE
Using rtx on the server

### DISCUSSION THREAD

**Discussion by GZLiew** [jdx/mise#869] [General]
Hi 

Im currently running asdf on a server but I had to do a lot of hoops and loops in order to work

not sure if rtx will solve these for me?

my setup

```bash
ASDF_DIR=/var/lib/buildkite-agent/.asdf
ASDF_DATA_DIR="${ASDF_DIR}"

# shellcheck disable=SC2120
function install_asdf() {
	local asdf_version=${1:-0.11.1}

	echo "--- Cloning asdf"

	sudo -u buildkite-agent bash <<-EOF
		git clone https://github.com/asdf-vm/asdf.git "${ASDF_DIR}" --branch v${asdf_version}
	EOF

	echo "--- Export variables"

	export HOME=~

	echo "--- Adding asdf to PATH"

	cat <<-EOF >/usr/bin/asdf
		#!/bin/bash
		export ASDF_DIR=${ASDF_DIR}
		export ASDF_DATA_DIR=${ASDF_DATA_DIR}
		source ${ASDF_DATA_DIR}/asdf.sh
		asdf "\$@"
	EOF

	echo "--- Make asdf executable"

	sudo chmod +x /usr/bin/asdf

	echo "Print asdf version"
	asdf --version
}

if ! command -v asdf &>/dev/null; then
	echo "asdf not found. Installing"
	install_asdf
fi

```

**Comment by Marukome0743** [comment]
mise should simplify this setup because it is distributed as a single executable and does not require cloning or sourcing a framework repository.

The important part is to run installation commands as the same account that runs the Buildkite jobs and keep its data directory writable by that account. For example:

```sh
sudo -H -u buildkite-agent env \
  MISE_DATA_DIR=/var/lib/buildkite-agent/.local/share/mise \
  mise install
```

For non-interactive build steps, shell activation is usually unnecessary. The most reliable option is:

```sh
mise exec -- your-command
```

This loads the tools and environment from `mise.toml` for that command.

Alternatively, the Buildkite agent environment can use mise shims:

```sh
export MISE_DATA_DIR=/var/lib/buildkite-agent/.local/share/mise
export PATH="/usr/local/bin:$MISE_DATA_DIR/shims:$PATH"
```

After installing tools, `mise reshim` can be used to rebuild the shims when necessary.

This avoids the wrapper function and `asdf.sh` sourcing shown in the original example. The `mise` binary can live in `/usr/local/bin`, while tool installations remain owned by the `buildkite-agent` account.

The current CI guidance is documented here:

https://mise.en.dev/continuous-integration.html

*AI-assisted — Tool: Codex; model: OpenAI/GPT-5; version: unavailable.*
